File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
  array = new Array(n+1).fill(true);
  count = 0;

  for (i of section) {
    array[i] = false;
  }

  for (i = 0; i < array.length; i++) {
    if (array[i] === false) {
      count++;
      array[i] = true;
      for (j = 0; j < m; j++) {
        if (i + j <= n) {
          array[i+j] = true;
        }
      }
    }
  }

  return count;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section): #전체길이, 페인트 길이, 벗겨진 부분
    
    array = [True] * (n+1)
    count = 0
    
    for i in section:
        array[i] = False
        
    for i in range(0,len(array)):
        if(array[i] == False):
            count = count + 1
            array[i] = True
            for j in range(m):
                if i + j <= n:
                    array[i+j] = True
                
    return count